### PR TITLE
PE-432 - Adding the activity completion report.

### PR DESCRIPTION
**Pull request by Squirrel18**
## Description:

This PR adds the activity completion report.

## Settings:

```
"ACTIVITY-COMPLETION-REPORT": {
        "REPORT_URL": "/proversity-reports/api/v0/generate-activity-completion-report",
        "BACKEND_REPORT": "proversity_reports_script.report_backend.activity_completion_report:ActivityCompletionReportBackend",
        "EXTRA_REQUEST_DATA": {
            "required_activity_ids": [
                "block-v1:edX+CS101+2014_T1+type@problem+block@0229689960e54520b33575a99e4343f4"
            ],
            "block_types": [
                "problem"
            ]
        },
        "EXTRA_DATA": {
            "BUCKET_NAME": "S3-bucket-name"
        }
    }
```

## Reviewers:

- [ ] @andrey-canon

### CODE REVIEW

File: proversity_reports_script/report_backend/activity_completion_report.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    Return a csv dict to write the csv file.
+    """
+    csv_data = []
+
+    for user_data in course_data:
+        email = user_data.get('email', '')
+        first_name = user_data.get('first_name', '')
+        last_name = user_data.get('last_name', '')
+
+        dict_writer_data = {}
```
Comment (andrey-canon) [inline proversity_reports_script/report_backend/activity_completion_report.py]: ordereddict ?

Comment (Squirrel18) [reply proversity_reports_script/report_backend/activity_completion_report.py]: Yesssssssss.

File: proversity_reports_script/report_backend/activity_completion_report.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    if total_activities:
+        total = int(total_activities)
+        # Create as many 'required_activity_' as the total number of activities.
+        for activity_number in range(1, total + 1): # Plus 1, because the stop argument it's not inclusive.
+            required_activity_number = user_data.get('required_activity_{}'.format(activity_number), '')
+            required_activity_name = user_data.get('required_activity_{}_name'.format(activity_number), '')
+
+            # Let's add the activity number at the end of the name if two or more activities have the same name.
+            if required_activity_name in required_activities_data.keys():
+                required_activity_name = '{}-{}'.format(
```
Comment (andrey-canon) [inline proversity_reports_script/report_backend/activity_completion_report.py]: one line

File: proversity_reports_script/report_backend/activity_completion_report.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        dict_writer_data['Email'] = email
+
+        dict_writer_data.update(get_required_activity_dict(user_data))
+
+        csv_data.append(dict_writer_data)
+
+    return csv_data
+
+
+def get_required_activity_dict(user_data):
```
Comment (andrey-canon) [inline proversity_reports_script/report_backend/activity_completion_report.py]: You are using this method in video_completion report, please extract this to a common file.

**Review by andrey-canon — CHANGES_REQUESTED** [review]

File: proversity_reports_script/report_backend/activity_completion_report.py (hunk 3 above)
Comment (Squirrel18) [reply proversity_reports_script/report_backend/activity_completion_report.py]: Good catch.

File: proversity_reports_script/report_backend/activity_completion_report.py (hunk 2 above)
Comment (Squirrel18) [reply proversity_reports_script/report_backend/activity_completion_report.py]: Changed.

**Comment by Squirrel18** [comment]
PR addressed. @andrey-canon

**Review by andrey-canon — APPROVED** [review]

**Pull request merged**